src/SFML/Graphics/Shader.cpp

```diff
@@ -896,16 +894,16 @@ bool Shader::compile(const char* vertexShaderCode, const char* geometryShaderCod
         glCheck(GLEXT_glGetInfoLog(shaderProgram, sizeof(log), nullptr, log));
         err() << "Failed to link shader:" << '\n' << log << std::endl;
         glCheck(GLEXT_glDeleteObject(shaderProgram));
-        return false;
+        return std::nullopt;
     }
 
-    m_shaderProgram = castFromGlHandle(shaderProgram);
+    Shader shader(castFromGlHandle(shaderProgram));
 
     // Force an OpenGL flush, so that the shader will appear updated
     // in all contexts immediately (solves problems in multi-threaded apps)
     glCheck(glFlush());
 
-    return true;
+    return shader;
```

Comment:
`castFromGlHandle` is just a cast, so you can: 

```cpp
// Force an OpenGL flush, so that the shader will appear updated
// in all contexts immediately (solves problems in multi-threaded apps)
glCheck(glFlush());

return Shader(castFromGlHandle(shaderProgram));
```